pkg/solana/keys/key.go

```diff
@@ -0,0 +1,110 @@
+package keys
+
+import (
+	"crypto"
+	"crypto/ed25519"
+	crypto_rand "crypto/rand"
+	"fmt"
+	"io"
+
+	"github.com/gagliardetto/solana-go"
+	"github.com/mr-tron/base58"
+)
+
+// Raw represents the ETH private key
+type Raw []byte
+
+// Key gets the Key
+func (raw Raw) Key() Key {
+	privKey := ed25519.NewKeyFromSeed(raw)
+	pubKey := make([]byte, ed25519.PublicKeySize)
+	copy(pubKey, privKey[ed25519.PublicKeySize:])
+	return Key{
+		privkey: privKey,
+		pubKey:  pubKey,
+	}
+}
+
+// String returns description
+func (raw Raw) String() string {
+	return "<Solana Raw Private Key>"
+}
+
+// GoString wraps String()
+func (raw Raw) GoString() string {
+	return raw.String()
+}
+
+var _ fmt.GoStringer = &Key{}
+
+// Key represents Solana key
+type Key struct {
```

Comment:
I was not expecting the keystore _implementation_ to be moved. What is the reason for this?

Reply:
StarkNet is also set up this way - so that might be why?

if I understand correctly, the keystore itself hasn't moved just the base implementation of the solana key

Reply:
But this means that the chainlink repo has to import this module, with all of the chain-specific dependencies, right? We'll just have to move it back eventually in that case.

Reply:
ahhhh right - this gets weird with plugins because we don't want core importing the relay or the relay importing core

Reply:
FWIW this is another reason to only use the minimal interface required by the txm, so that all of this other stuff doesn't even come up.

Reply:
yeah that makes sense now - I didn't think this far down the line

so that minimal interface needs to live in `chainlink-relay` where `chainlink-solana` doesn't need direct knowledge of the the solana keystore that lives in core

Reply:
i'm debating if this should be done in a follow up PR 🤔

Reply:
you should be able to keep it even simpler for now, w/o a common interface in relay just yet. but I'm hacking on one now for reference:
https://github.com/smartcontractkit/chainlink-relay/pull/80/files#diff-84405d57992ad70fad77392b91f683b092a624afe5037a279f362b21474f4dfbR36-R39

Reply:
(don't worry about context)

Reply:
makes sense, so on the `chainlink-solana` side we can define a simple keystore interface that provides a sign method with inputs (context, string address, byte data) and outputs (bytes signature, error)

then on the core side, add an additional keystore method that matches

how should the keystore respond if the key doesn't exist? I'm thinking just throw an error

Reply:
You can wrap the existing pretty trivially:
https://github.com/smartcontractkit/chainlink/pull/8517/files#diff-29f963c1e6175aceb3883554c8a75d78e3fc565dbd374ec1daf54d5036c0bc58R25-R35

Reply:
I think preserving error behavior for now is simplest, but we may have to adjust it later (and I'll be sure to document what the plugin API expects).

Reply:
alrighty, I will handle this in a PR built on this one and make updates :)

Reply:
resolving this conversation for now since a simplified keystore interface is now used based on previous discussions